masschallenge/impact-api
# [AC-4900] Update PATCH and POST for User

To test:
- `make load-remote-db DB_FILE_NAME=initial_schema.sql.gz`
- `make grant-permissions PERMISSION_USER=demoadmin@masschallenge.org PERMISSION_CLASSES=v1_clients,v1_confidential`
- `make dev`
- Login as demoadmin@masschallenge.org
- Do a POST to /v1/user with various parameters as described in the ticket.  See that you get what you expect.
- Do various PATCHes to different fields of /v1/user/<id> and see that it does what you expect.
- There should a reasonable level of error checking on the various fields.

Note: The PATCH related work is relatively well organized with appropriately shared validators and a shared setting mechanism.  The POST work leverages some of this, but could definitely be better integrated.  However, the OPTIONS are still haphazard.  Improvements to OPTIONS are planned in [AC-4857](https://masschallenge.atlassian.net/browse/AC-4857) as is a more general re-architecture that will bring GET, POST, PATCH and OPTIONS much closer together.  Given that this is a timeboxed ticket, we will push back on any requests to refactor POST to leverage more of the PATCH infrastructure.
